### PR TITLE
Print additional log for failing event-replay

### PR DESCRIPTION
**Pull request by MHassanTariq**
## Description

This PR adds an additional error log specifying failing event-replay to help automatically detect such failure.  

For details refer to issue #944 

## Type of Change
- [ ] New feature
- [ ] Bug fix
- [ ] API reference/documentation update
- [X] Other

## Checklist
- [ ] Code is commented where needed
- [ ] Unit test coverage for new or modified code paths
- [ ] `npm run test` passes
- [ ] Changelog is updated
- [ ] Tag 1 of @rafaelcr or @zone117x for review

### CODE REVIEW

File: src/event-stream/event-server.ts [hunk 1]
```diff
@@ -817,6 +817,7 @@ export async function startEventServer(opts: {
         res.status(200).json({ result: 'ok' });
       } catch (error) {
         logError(`error processing core-node /new_block: ${error}`, error);
+        logError(`Event-replay process failed due to the aforementioned error`);
```
Comment (rafaelcr) [inline src/event-stream/event-server.ts:820]: These errors could happen also if there wasn't an event replay in progress, so I don't think these messages should be printed here.

IIRC, there's an `eventReplay` property in the `index.ts` file that shows whether or not we're actually performing one, so I think this logic should be handled over there.

Comment (MHassanTariq) [reply src/event-stream/event-server.ts]: Thanks for the help on this @rafaelcr 

I went through the documentation for [Event-Replay](https://github.com/hirosystems/stacks-blockchain-api#event-replay) and noticed that event-replay initiates on `export-events` or `import-events` commands are given. Therefore, I've logged the error in the catch. Can you check it once and let me know if that looks good to you? 

cc: @CharlieC3

File: src/index.ts [hunk 2]
```diff
@@ -381,5 +381,6 @@ async function handleProgramArgs() {
 
 void handleProgramArgs().catch(error => {
   console.error(error);
+  console.error(`Event-replay process failed due to the aforementioned error`);
```
Comment (rafaelcr) [inline src/index.ts:384]: You should only show this error when the command arg was `import-events`

**Review by rafaelcr — CHANGES_REQUESTED** [review]

File: src/index.ts (hunk 2 above)
Comment (rafaelcr) [reply src/index.ts:384]: Alternatively, you can concat the operand to this string so we have an error that would work for all operations e.g.:
```js
`${args.operand} process failed`
```

File: src/index.ts [hunk 3]
```diff
@@ -274,7 +274,7 @@ function initApp() {
     });
 }
 
-async function handleProgramArgs() {
+function getEventReplayArgs() {
```
Comment (rafaelcr) [inline src/index.ts:277]: Change name to `getProgramArgs`.
Remember this function parses the args not only for event replay but for event export, etc.

**Review by rafaelcr — APPROVED** [review]
🏁

File: src/index.ts [hunk 4]
```diff
@@ -381,5 +385,7 @@ async function handleProgramArgs() {
 
 void handleProgramArgs().catch(error => {
   console.error(error);
+  const { args } = getProgramArgs();
+  console.error(`${args.operand} process failed`);
```
Comment (zone117x) [inline src/index.ts:389]: This causes the message `undefined process failed` to be printed when the API exits from an error when started regularly e.g. `node ./lib/index.js`

**Review by rafaelcr — CHANGES_REQUESTED** [review]
My bad, missed @zone117x 's observation. Can you fix it please @MHassanTariq ?

File: src/index.ts (hunk 4 above)
Comment (MHassanTariq) [reply src/index.ts:389]: In that case, it won't go for events and [initiate the application](https://github.com/hirosystems/stacks-blockchain-api/blob/master/src/index.ts#L331) instead. That has its own [error handler](https://github.com/hirosystems/stacks-blockchain-api/blob/master/src/index.ts#L219-L228) which exits the program. 
Please let me know if I'm wrong here.

Comment (zone117x) [reply src/index.ts:389]: Possibly, there's async and sync code mixed in here. Better to just check for null/undefined and avoid logging

**Review by zone117x — APPROVED** [review]
LGTM!

**Pull request merged**